### PR TITLE
【NPU】Support npu op elementwise_div and elementwise_div_grad

### PR DESCRIPTION
**Pull request by oyjxer**
### PR types
New features

### PR changes
OPs

### Describe
【NPU】Support npu op elementwise_div and elementwise_div_grad

单测结果：
elementwise_div success

调用elementwise_div npu kernel：
elementwise_div forward

调用elementwise_div_grad npu kernel：
elementwise_div backward

### CODE REVIEW

File: paddle/fluid/operators/elementwise/elementwise_div_op_npu.cc [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    auto* x = ctx.Input<Tensor>("X");
+    auto* y = ctx.Input<Tensor>("Y");
+
+    auto* dx = ctx.Output<Tensor>(framework::GradVarName("X"));
+    auto* dy = ctx.Output<Tensor>(framework::GradVarName("Y"));
+
+    auto place = ctx.GetPlace();
+
+    dx->mutable_data<T>(place);
+    dy->mutable_data<T>(place);
```
Comment (liym27) [inline paddle/fluid/operators/elementwise/elementwise_div_op_npu.cc]: Please consider cases where the `dx==nullptr` or `dy==nullptr`, and test them in `test_elementwise_div_op_npu.py`. Please refer https://github.com/PaddlePaddle/Paddle/pull/31347

And fix code of `ElementwiseMulGradNPUKernel` in https://github.com/PaddlePaddle/Paddle/pull/31571 for the same reason.

Comment (oyjxer) [reply paddle/fluid/operators/elementwise/elementwise_div_op_npu.cc]: fix

File: paddle/fluid/operators/elementwise/elementwise_div_op_npu.cc [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    if (dx) {
+      dx->mutable_data<T>(place);
+      auto x_grad_runner = NpuOpRunner("Mul", {x_grad_w, *dout}, {*dx}, {});
+      x_grad_runner.Run(stream);
+    }
+
+    if (dy) {
+      dy->mutable_data<T>(place);
+      auto y_grad_runner = NpuOpRunner("Mul", {y_grad_w, *dout}, {*dy}, {});
+      y_grad_runner.Run(stream);
```
Comment (zhiqiu) [inline paddle/fluid/operators/elementwise/elementwise_div_op_npu.cc]: Do not need to do previous operations if dx or dy is nullptr.

Comment (oyjxer) [reply paddle/fluid/operators/elementwise/elementwise_div_op_npu.cc]: previous operations not need dx or dy

Comment (oyjxer) [reply paddle/fluid/operators/elementwise/elementwise_div_op_npu.cc]: fix

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**